satpy/dependency_tree.py

```diff
@@ -441,7 +441,7 @@ def _promote_query_to_modified_dataid(self, query, dep_key):
 
     def get_compositor(self, key):
         """Get a compositor."""
-        for sensor_name in self.compositors.keys():
+        for sensor_name in sorted(self.compositors.keys()):
```

Comment:
Why the sorting? Is it tested?

Reply:
See my description at the top. It isn't tested as far as I know and is a lightly used feature at best (supporting multiple sensors in a single Scene). I should add a test.